fbw-a32nx/src/systems/fmgc/src/flightplanning/RawDataMapper.ts

```diff
@@ -60,11 +61,36 @@ export class RawDataMapper {
                 approach.longName = ApproachUtils.longApproachName(approach)
             });
 
-            info.approaches.forEach(
-                (approach) => approach.transitions.forEach(
-                    (trans) => trans.name.trim().length === 0 && (trans.name = WayPoint.formatIdentFromIcao(trans.legs[0].fixIcao)),
-                ),
-            );
+            info.approaches.forEach((approach) => {
+                approach.transitions.forEach((trans) => {
+                    // if the trans name is empty (in some 3pd navdata), fill it with the IAF name
+                    if (trans.name.trim().length === 0) {
+                        trans.name = WayPoint.formatIdentFromIcao(trans.legs[0].fixIcao)
+                    }
+
+                    // Fix up navigraph approach transitions which hide IAPs inside other transitions rather
+                    // than splitting them out as they should be in MSFS data. Unfortunately this means
+                    // these transitions cannot be synced to the sim's flight plan system for ATC etc. as
+                    // they're not visible without this hack.
+                    // Note: it is safe to append to the array inside the forEach by the ECMA spec, and the appended
+                    // elements will not be visited.
+                    for (let i = 1; i < trans.legs.length; i++) {
+                        const leg = trans.legs[i];
+                        if ((leg.fixTypeFlags & FixTypeFlags.IAF) > 0 && (leg.type === LegType.TF || leg.type === LegType.IF)) {
```

Comment:
``(leg.type === LegType.TF || leg.type === LegType.IF)``

Perhaps could be made into its own codeblock with a ``switch case`` instead which would be slightly more readable?
Although this also works and is understandable as is.

Reply:
I'd agree for more options, but 2 is more readable like this.